xcodeproject/xcodeproj/target.go

```diff
@@ -31,49 +31,16 @@ type Target struct {
 	buildPhaseIDs          []string
 }
 
-// DependentTargets ...
-func (t Target) DependentTargets() []Target {
-	var targets []Target
+// DependsOn ...
+func (t Target) DependsOn(targetID string) bool {
 	for _, targetDependency := range t.Dependencies {
-		childTarget := targetDependency.Target
-		targets = append(targets, childTarget)
-
-		childDependentTargets := childTarget.DependentTargets()
-		targets = append(targets, childDependentTargets...)
-	}
-
-	return targets
-}
-
-// DependesOn ...
-func (t Target) DependesOn(targetID string) bool {
-	for _, targetDependency := range t.Dependencies {
-		childTarget := targetDependency.Target
-		if childTarget.ID == targetID {
+		if targetDependency.TargetID == targetID {
 			return true
 		}
 	}
 	return false
 }
 
-// DependentExecutableProductTargets ...
-func (t Target) DependentExecutableProductTargets() []Target {
```

Comment:
This was only used at 2 places, so I decided to inline at the callsites. This method was basically the same as `DependentTargets()` with an extra filtering.

Reply:
Good point! But isn't the old implementation (potentially) incorrect because it skips certain targets, while the new implementation returns with all the possible executable targets?

I'm fine with re-adding this method to keep behavior the same, but I don't see any harm in this behavior change as it returns more, not fewer targets.

Reply:
After some discussion I now think it is a good idea to deduplicate the recursive iteration; as we want to make it more robust in the next PR (to reduce extraneous iterations).
There is no obvious use case where new executable targets would be returned even with the changed logic, so we can go ahead with this.